vizro-core/src/vizro/managers/_model_manager.py

```diff
@@ -50,90 +50,58 @@ def __iter__(self) -> Generator[ModelID, None, None]:
 
         Note this yields model IDs rather key/value pairs to match the interface for a dictionary.
         """
+        # TODO: should this yield models rather than model IDs? Should model_manager be more like set with a special
+        #  lookup by model ID or more like dictionary?
         yield from self.__models
 
-    # TODO: Consider adding an option to iterate only through specific page - "in_page_with_id=None"
-    def _items_with_type(self, model_type: type[Model]) -> Generator[tuple[ModelID, Model], None, None]:
-        """Iterates through all models of type `model_type` (including subclasses)."""
-        for model_id in self:
-            if isinstance(self[model_id], model_type):
-                yield model_id, cast(Model, self[model_id])
-
-    # TODO: Consider returning with yield
-    # TODO: Make collection of model ids (throughout this file) to be set[ModelID].
-    def _get_model_children(self, model_id: ModelID, all_model_ids: Optional[list[ModelID]] = None) -> list[ModelID]:
-        if all_model_ids is None:
-            all_model_ids = []
-
-        all_model_ids.append(model_id)
-        model = self[model_id]
-        if hasattr(model, "components"):
-            for child_model in model.components:
-                self._get_model_children(child_model.id, all_model_ids)
-        if hasattr(model, "tabs"):
-            for child_model in model.tabs:
-                self._get_model_children(child_model.id, all_model_ids)
-        return all_model_ids
-
-    # TODO: Consider moving this method in the Dashboard model or some other util file
-    def _get_model_page_id(self, model_id: ModelID) -> ModelID:  # type: ignore[return]
+    def _get_models(
+        self, model_type: Optional[Union[type[Model], tuple[type[Model], ...]]] = None, page: Optional[Page] = None
+    ) -> Generator[Model, None, None]:
+        """Iterates through all models of type `model_type` (including subclasses).
+
+        If `model_type` not given then look at all models. If `page` specified then only give models from that page.
+        """
+        models = self._get_model_children(page) if page is not None else self.__models.values()
+
+        for model in models:
+            if model_type is None or isinstance(model, model_type):
+                yield model
+
+    def _get_model_children(self, model: Model) -> Generator[Model, None, None]:
+        from vizro.models import VizroBaseModel
+
+        if isinstance(model, VizroBaseModel):
+            yield model
+
+        model_fields = ["components", "tabs", "controls", "actions", "selector"]
```

Comment:
I would recommend putting this in a constant. I know this is not your style if only used once, but this should be more visible so that later updating becomes clearer

Reply:
This I have decided not to do because I don't intend to update this list later - instead I'd like it to become completely obsolete as part of `model_manager` refactoring where we can look at _all_ models recursively without listing explicitly the fields we look at. I'll add a TODO to explain this.

Reply:
[90d5868](https://github.com/mckinsey/vizro/pull/903/commits/90d586813fc33ffc28f272f1a27e7c7904b72a8d)